src/porepy/numerics/ad/operators.py

```diff
@@ -447,24 +525,20 @@ def _parse_operator(self, op: Operator, mdg: pp.MixedDimensionalGrid):
                 raise ValueError(msg) from exc
 
         elif operation == Operator.Operations.evaluate:
-            # This is a function, which should have at least one argument
-            assert len(results) > 1
-            func_op = results[0]
+            # Operator functions should have at least 1 child (themselves)
+            assert len(results) >= 1, "Operator functions must have at least 1 child."
+            assert hasattr(op, "func"), (
+                f"Operators with operation {operation} must have a functional"
+                + f" representation `func` implemented as a callable member."
+            )
 
-            # if the callable can be fed with AdArrays, do it
```

Comment:
The mess of this uneccessary flags is removed.
The `Operation.evaluate` now checks that we are dealing with operator functions and that `func` is available.
Notice that Operator functions do not require themselves as children anymore, which shortens the recursion tree a bit.